In [11]:
import numpy as np
from scipy import stats
from math import sqrt

In [12]:
# Data
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36,
                 1.32, 0.85, 1.07, 1.48, 1.20, 1.33,
                 1.18, 1.22, 1.29])

In [13]:
# Basic statistics
n = len(data)
xbar = data.mean()
s = data.std(ddof=1)   # sample SD (unbiased)
alpha = 0.01
df = n - 1

In [14]:
print("Sample size (n):", n)
print("Sample mean (x̄):", round(xbar, 6))
print("Sample SD (s):", round(s, 6))

Sample size (n): 15
Sample mean (x̄): 1.238667
Sample SD (s): 0.193164


# (a) 99% CI using sample SD (t-distribution)

In [15]:
se_s = s / sqrt(n)
tcrit = stats.t.ppf(1 - alpha/2, df)
moe_a = tcrit * se_s
ci_a = (xbar - moe_a, xbar + moe_a)
print("\n(a) 99% CI using t-distribution (σ unknown):", tuple(round(v, 6) for v in ci_a))


(a) 99% CI using t-distribution (σ unknown): (np.float64(1.090197), np.float64(1.387136))


# (b) 99% CI using known population SD (z-distribution)

In [16]:
sigma = 0.2
se_sigma = sigma / sqrt(n)
zcrit = stats.norm.ppf(1 - alpha/2)
moe_b = zcrit * se_sigma
ci_b = (xbar - moe_b, xbar + moe_b)
print("(b) 99% CI using z-distribution (σ=0.2):", tuple(round(v, 6) for v in ci_b))

(b) 99% CI using z-distribution (σ=0.2): (np.float64(1.105651), np.float64(1.371682))


# Conclusion

#### The print-head durability study

Based on 15 destructively tested print-heads, the sample mean durability is ≈ 1.239 million characters.

With σ unknown (the realistic case here), the 99% t-interval is (1.090, 1.387) million.

If the population σ were known to be 0.2, the 99% z-interval would be (1.106, 1.372) million—slightly narrower, as expected.